src/System.Private.ServiceModel/tests/Scenarios/Contract/Service/ServiceContractTests.cs

```diff
@@ -672,4 +672,233 @@ public static void ServiceContract_Call_Operation_With_MessageParameterAttribute
             ScenarioTestHelpers.CloseCommunicationObjects((ICommunicationObject)serviceProxy, factory);
         }
     }
+    
+    [Fact]
+    [OuterLoop]
+    public static void BasicHttp_Abort_ChannelFactory_Operations_Active()
+    {
+        // Test creates 2 channels from a single channel factory and
+        // aborts the channel factory while both channels are executing
+        // operations.  This verifies the operations are cancelled and
+        // the channel factory is in the correct state.
+        BasicHttpBinding binding = null;
+        TimeSpan delayOperation = TimeSpan.FromSeconds(3);
+        ChannelFactory<IWcfService> factory = null;
+        IWcfService serviceProxy1 = null;
+        IWcfService serviceProxy2 = null;
+
+        try
+        {
+            // *** SETUP *** \\
+            binding = new BasicHttpBinding(BasicHttpSecurityMode.None);
+            binding.CloseTimeout = ScenarioTestHelpers.TestTimeout;
+            factory = new ChannelFactory<IWcfService>(binding, new EndpointAddress(Endpoints.HttpBaseAddress_Basic));
+            serviceProxy1 = factory.CreateChannel();
+            serviceProxy2 = factory.CreateChannel();
+
+            // *** EXECUTE *** \\
+            Task<string> t1 = serviceProxy1.EchoWithTimeoutAsync("first", delayOperation);
+            Task<string> t2 = serviceProxy2.EchoWithTimeoutAsync("second", delayOperation);
+            factory.Abort();
+
+            // *** VALIDATE *** \\
+            Assert.True(factory.State == CommunicationState.Closed,
+                        String.Format("Expected factory state 'Closed', actual was '{0}'", factory.State));
+
+            Assert.Throws<TaskCanceledException>(() => t1.GetAwaiter().GetResult());
```

Comment:
This doesn't check that the task is cancelled when factory.Abort() returns as t1.GetAwaiter().GetResult() might block waiting for the task to complete if we have a problem where we actually wait for the operation to complete and then throw TaskCancelledException. Instead use `Assert.True(t1.IsCanceled);`